File: easy/sum_digits.php
<?php
$fh = fopen($argv[1], "r");
$c = 0;
while (true) {
	$test = fgetc($fh);

	if ($test == "") {
		break;
	}
	elseif ($test == "\n") {
		echo $c, "\n";
		$c = 0;
	}
	else {
		$c += intval($test);
	}
}
?>

File: easy/sum_digits.c
#include <stdio.h>

int main(int argc, char *argv[]) {
	FILE *fp;
	int n = 0;
	char c;

	fp = fopen(*++argv, "r");
	while ((c = getc(fp)) != EOF || n > 0)
		if (c == '\n' || c == EOF) {
			printf("%d\n", n);
			n = 0;
		} else {
			n += c - '0';
		}
	return 0;
}
